readthedocs/vcs_support/backends/git.py

```diff
@@ -68,6 +91,104 @@ def update(self):
+            return f"refs/tags/{self.verbose_name}:refs/tags/{self.verbose_name}"
+
+        if self.version_type == EXTERNAL:
+            # TODO: We should be able to resolve this without looking up in oauth registry
+            git_provider_name = self.project.git_provider_name
+
+            # Remote reference for Git providers where pull request builds are supported
+            if git_provider_name == GITHUB_BRAND:
+                return GITHUB_PR_PULL_PATTERN.format(id=self.verbose_name)
+            if self.project.git_provider_name == GITLAB_BRAND:
+                return GITLAB_MR_PULL_PATTERN.format(id=self.verbose_name)
+
+    def clone_ng(self):
+        # If the repository is already cloned, we don't do anything.
+        # This is legacy from when cached the repository on disk,
+        # but we may want to be able to call .update()
+        # several times in the same build
+        if self.repo_exists():
+            return
+
+        # This seems to be required for test cases to work
+        self.make_clean_working_dir()
+
+        # --no-checkout: Makes it explicit what we are doing here. Nothing is checked out
+        #                until it's explicitly done.
+        # --depth 1: Shallow clone, fetch as little data as possible.
+        cmd = ["git", "clone", "--no-checkout", "--depth", "1", self.repo_url, "."]
+        try:
+            # TODO: Explain or remove the return value
+            code, stdout, stderr = self.run(*cmd)
+            return code, stdout, stderr
+        except RepositoryError as exc:
+            raise RepositoryError(RepositoryError.CLONE_ERROR()) from exc
+
+    def fetch_ng(self):
+        """Implementation for new clone+fetch+checkout pattern."""
+
+        # --force: Likely legacy, it seems to be irrelevant to this usage
+        # --prune: Likely legacy, we don't expect a previous fetch command to have run
+        # --prune-tags: Likely legacy, we don't expect a previous fetch command to have run
+        # --tags: This flag was used in the previous approach such that all tags were fetched
+        #         in order to checkout a tag afterwards.
+        # --depth: This flag should be made unnecessary, it's downloading commit data that's
+        #          never used.
+        cmd = [
+            "git",
+            "fetch",
+            "origin",
+            "--force",
+            "--prune",
+            "--prune-tags",
+            "--depth",
+            str(self.repo_depth),
+        ]
+        remote_reference = self.get_remote_fetch_reference()
+
+        if remote_reference:
+            # TODO: We are still fetching the latest 50 commits.
+            # A PR might have another commit added after the build has started...
+            cmd.append(remote_reference)
+        else:
+            # We are doing a fetch without knowing the remote reference.
+            # This is expensive, so log the event.
+            log.info(
```

Comment:
```suggestion
            log.warning(
```

Reply:
Will get back to this. Just checking out the possible known cases for this scenario.

Reply:
Firstly, I converted it to a warning.

But I've since learned that we cannot currently know anything about the remote refspec for Version "stable" because we only store the commit ID but no info about its origin refspec (either tag or branch name) :/

Reply:
Keeping it as a warning but only logging if it's not for a "stable" version.